File: Day 1/java/level 1/LevelOneTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

	public class LevelOneTest {

	@Test
	public void testThatFunctionSplitsArrayIntoEvenAndOddArrays(){

		int[] array = {1,2,3,4,5,6};

		int[][] expected = {{2,4,6},{1,3,5}};

		int[][] actual = LevelOne.arraySplitter(array);
		
		assertArrayEquals(actual, expected);



}
		
	@Test
	public void testThatFunctionChecksIfArrayIsPallindrome(){

		int[] array = {4,6,4};

		boolean expected = true;

		boolean actual = LevelOne.palindromicArray(array);
		
		assertEquals(actual, expected);
}


	@Test
	public void testThatFunctionReturnsPerfectSquare(){

		int[] array = {3,4,5,9,12,16};
	
		int[] expected = {4,9,16};

		int[] actual = LevelOne.getPerfectSquare(array);
		
		assertArrayEquals(actual, expected);

}



	@Test 
	public void testthatFunctionReplacesNonPerfectSquaresWithNegativeOne(){

		int[] array = {7,4,3,5,9,16,21};

		int[] expected = {-1,4,-1,-1,9,16,-1};

		int[] actual  = LevelOne.replaceNonPerfectSquare(array);

		assertArrayEquals(expected, actual);



}































}

File: Day 1/python/level 1/LevelOneTest.py
from level_one import *


def test_array_splitter():

    array = [1, 2, 3, 4, 5, 6]

    expected = [[2, 4, 6], [1, 3, 5]]

    actual = array_splitter(array)

    assert actual == expected


def test_palindromic_array():

    array = [45, 0, 8, 0, 45]

    expected = True

    actual = palindromic_array(array)

    assert actual == expected


def test_get_perfect_square():

    array = [4, 7, 9, 10, 16, 18]

    expected = [4, 9, 16]

    actual = get_perfect_square(array)

    assert actual == expected


def test_replace_non_perfect_square():

    array = [4, 7, 9, 10, 16, 18]

    expected = [4, -1, 9, -1, 16, -1]

    actual = replace_non_perfect_square(array)

    assert actual == expected